webpush/encrypt.go

```diff
@@ -139,36 +142,64 @@ type EncryptionResult struct {
 	ServerPublicKey []byte
 }
 
+// Decrypt an encrypted messages.
```

Comment:
It would make it a bit tricky as API - I can move the code to a separate file, and keep it only in my fork. Having 2 packages would require exposing too much. I'm trying to make it complete - i.e. including delivery receipts, which involves http2 framer, UA receipt side with push promises, and the basic elements of the server. If this doesn't fit this repo, I can organize the code to keep the extra functionality in separate files in a fork.